src/Workspaces/CSharp/Portable/Simplification/Simplifiers/NameSimplifier.cs

```diff
@@ -510,8 +517,18 @@ private static bool CanSimplifyNullable(INamedTypeSymbol type, NameSyntax name,
                                 newAttributeName,
                                 identifierToken.TrailingTrivia));
 
-                        replacementNode = SyntaxFactory.IdentifierName(newIdentifierToken)
-                            .WithLeadingTrivia(name.GetLeadingTrivia());
+                        switch (name)
```

Comment:
canthis function be updated to take a SimpleNameSYntax instead of a TYpeSyntax.

Reply:
It does not seem like that would be any helpful; the caller assigns this into an `out TypeSyntax` parameter in an override method. I don't see this being worth the hassle.